plugins/modules/ec2_instance.py

```diff
@@ -922,6 +922,12 @@ def build_volume_spec(params):
             for int_value in ['volume_size', 'iops']:
                 if int_value in volume['ebs']:
                     volume['ebs'][int_value] = int(volume['ebs'][int_value])
+            if 'volume_type' in volume['ebs'] and volume['ebs']['volume_type'] == 'gp3':
+                if not volume['ebs'].get('iops'):
+                    volume['ebs']['iops'] = 3000
+                if 'throughput' in volume['ebs']:
+                    volume['ebs']['throughput'] = int(volume['ebs']['throughput'])
```

Comment:
maybe set it also to its default value `125` if not set?  
what happens when the instance exist already? will it keeps the instance values (_in case they were changed manually_) or will it overwrite them (_reset them_)? I've no idea.

Reply:
Hi @markuman, Thank you for reviewing. I set the default value for `throughput`.

If my understanding is correct, you can specify different parameters for ebs like throughput and iops only when a new instance is created. In the case an EC2 instance is already present and you would like to modify throughput or iops or both you won’t be allowed because the API allows to modify only a specific subset of BlockDeviceMappings as shows here https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.modify_instance_attribute
I tried by myself to only change these params (throughput for example) and the task resulted in unchanged reflecting the documentation. Therefore, if you want to change throughput or iops values you would need to use the ec2_vol module.